src/components/timecard/edit-shift-hours/EditShiftHours.jsx

```diff
@@ -59,20 +62,37 @@ const EditShiftHours = ({
     }
   };
 
-  const handleServerValidationErrors = (error) => {
+  const handleServerValidationErrors = (errors) => {
+    if (errors === null || !Array.isArray(errors)) {
+      return false;
+    }
     const summaryErrors = {};
     let errorsHandled = true;
 
-    if (
-      error ==
-      ' has the following error(s): Time periods must not overlap with another time period'
-    ) {
-      summaryErrors['shift-start-time'] = {
-        message: 'Time periods must not overlap with another time period',
-      };
-    } else {
-      errorsHandled = false;
-    }
+    errors.forEach((error) => {
+      if (error.field === 'startAndEndTime') {
+        setClashingTimes(error.data);
+        setClashingProperty(error.field);
+        summaryErrors['shift-start-time'] = {
+          message:
+            'Your start and finish times must not overlap with another time period',
+        };
+      } else if (error.field === 'startTime') {
+        setClashingTimes(error.data);
+        setClashingProperty(error.field);
+        summaryErrors['shift-start-time'] = {
+          message: 'Your start time must not overlap with another time period',
+        };
+      } else if (error.field === 'endTime') {
+        setClashingTimes(error.data);
+        setClashingProperty(error.field);
+        summaryErrors['shift-finish-time'] = {
+          message: 'Your end time must not overlap with another time period',
+        };
```

Comment:
Is it possible to overlap with multiple finish times?
E.g. There exists 2 complete time entries on a Wednesday. I save a new time entry on e.g. Tuesday 11pm to Thursday 1am.

Technically i'd clash with those 2 finish times on Wednesday right?
If that is a valid scenario to display all those overlapping finish periods in the error message I think this would only show the last one.

But maybe that scenario isn't valid/relevant in which case it's fine. 
I guess my point is, are we sure we only have to ever account for a single clashing finish time?

Reply:
I'm not sure I follow? I've made a scenario where the finish time clashes with two other shift and it seems to work as intended, unless this isn't the situation you mean?
<img width="823" alt="Screenshot 2022-11-14 at 10 00 19" src="https://user-images.githubusercontent.com/113997250/201631644-0e1f283f-42da-4527-8841-fb05070bf5e3.png">

Reply:
Great, yeah I think that's the scenario I was curious about. Looks like it's all handled. Will resolve.